bricks/next-builder/src/search-tree/utils.tsx

```diff
@@ -0,0 +1,338 @@
+import { symbolForNodeId } from "../shared/storyboard/buildStoryboard";
+
+export const HIGHTLIGHT = "$$hightlight";
+export const NODE_INFO = "$$info";
+
+const iconTypeConstants = {
+  app: {
+    color: "cyan",
+    icon: "landmark",
+    lib: "fa",
+    prefix: "fas",
+  },
+  bricks: {
+    category: "app",
+    color: "cyan",
+    icon: "container",
+    lib: "easyops",
+  },
+  brick: {
+    color: "cyan",
+    icon: "codepen",
+    lib: "antd",
+    theme: "outlined",
+  },
+  customTemplates: {
+    color: "cyan",
+    icon: "superscript",
+    lib: "fa",
+    prefix: "fas",
+  },
+  i18n: {
+    color: "cyan",
+    icon: "globe-americas",
+    lib: "fa",
+    prefix: "fas",
+  },
+  slots: {
+    color: "cyan",
+    icon: "codepen-circle",
+    lib: "antd",
+    theme: "outlined",
+  },
+  routes: {
+    category: "app",
+    color: "cyan",
+    icon: "flow",
+    lib: "easyops",
+  },
+  redirect: {
+    category: "app",
+    color: "cyan",
+    icon: "flow",
+    lib: "easyops",
+  },
+  main: {
+    color: "cyan",
+    icon: "appstore-add",
+    lib: "antd",
+    theme: "outlined",
+  },
+  menus: {
+    color: "cyan",
+    icon: "list-ul",
+    lib: "fa",
+    prefix: "fas",
+  },
+  meta: {
+    color: "cyan",
+    icon: "cogs",
+    lib: "fa",
+    prefix: "fas",
+  },
+};
+
+const ingoreKey = ["bricks", "slots"];
+
+const supportKey = [
+  "routes",
+  "bricks",
+  "slots",
+  "meta",
+  "customTemplates",
+  "menus",
+  "i18n",
+  "app",
+  "zh",
+  "en",
+  "main",
+];
+
+let isParentRouteLock = false;
+
+export interface PlainObject extends Object {
+  [key: string]: any;
+}
+interface filterOption {
+  allowKeySearch?: boolean;
+}
+interface builTreeOptions {
+  parentPath?: string;
+  parentId?: string;
+  isSlots?: boolean;
+}
+
+function getType(item: PlainObject): keyof typeof iconTypeConstants {
+  if (item.brick) return "brick";
+
+  return item?.type;
+}
+
+function getTypeIcon(type: keyof typeof iconTypeConstants) {
+  if (iconTypeConstants[type]) {
+    return (
+      <GeneralIcon
+        style={{
+          fontSize: 12,
+          margin: "0 2px",
+        }}
+        icon={iconTypeConstants[type] as MenuIcon}
+      />
+    );
+  } else {
+    return (
+      <span
+        style={{
+          display: "inline-block",
+          width: 12,
+          height: 12,
+          margin: "0 2px",
+        }}
+      />
+    );
+  }
+}
+
+export function clone(obj: PlainObject) {
+  const newObj: PlainObject = Object.create(null);
+  for (const [k, v] of Object.entries(obj)) {
+    if (!ingoreKey.includes(k)) {
+      newObj[k] = v;
+    }
+  }
+  Object.getOwnPropertySymbols(obj).forEach((symbol) => {
+    newObj[symbol as any] = obj[symbol as any];
+  });
+  return newObj;
+}
+
+export function getTitle(item: PlainObject, defaultKey?: string): string {
+  return (
+    item?.menuId ||
+    item?.brick ||
+    item?.alias ||
+    item?.path ||
+    item?.name ||
+    item?.type ||
+    item?.title ||
+    defaultKey
```

Comment:
这会导致多次不必要的 `?.` 的转换，建议：

```suggestion
    (item && (
      item.menuId ||
      item.brick ||
      item.alias ||
      item.path ||
      item.name ||
      item.type ||
      item.title
    )) || defaultKey
```

Reply:
好的